service/worker/workerdeployment/version_workflow.go

```diff
@@ -591,7 +548,6 @@ func (d *VersionWorkflowRunner) handleSyncState(ctx workflow.Context, args *depl
 		// Version deactivated from current/ramping
 		d.VersionState.LastDeactivationTime = args.RoutingUpdateTime
 		d.startDrainage(ctx, false)
-		state.Status = enumspb.WORKER_DEPLOYMENT_VERSION_STATUS_DRAINING
 	}
 
 	// started accepting new workflows --> stop drainage child wf if it exists
```

Comment:
same here, you could compare previous status to `newStatus` instead of `if !wasAcceptingNewWorkflows && isAcceptingNewWorkflows`